guessing-game-v2/contracts/GuessingGame.sol

```diff
@@ -22,45 +22,49 @@ contract GuessingGame {
 
     constructor() {
         owner = msg.sender;
-        _resetSecretNumber();
+        _setSecretNumber();
     }
 
-    function guess(uint256 _number) external payable {
-        require(_number > 0 && _number <= MAX_GUESS, 'Secret number should be between 1 and 1000');
-        require(msg.value == GUESS_FEE, 'You need to pay a fees of 0.443 ETH to make a guess');
-        totalGuesses += 1;
+function guess(uint256 _number) external payable {
+    require(_number > 0 && _number <= MAX_GUESS, 'Secret number should be between 1 and 1000');
+    require(msg.value == GUESS_FEE, 'You need to pay a fees of 0.443 ETH to make a guess');
+    totalGuesses += 1;
 
-        if (lastGuess[msg.sender] == 0 || lastResetTime > lastGuess[msg.sender]) {
-            lastGuess[msg.sender] = 0;
-            lastDifference[msg.sender] = 0;
-        }
+    if (lastGuess[msg.sender] == 0 || lastResetTime > lastGuess[msg.sender]) {
+        lastDifference[msg.sender] = MAX_GUESS + 1;
+    }
```

Comment:
so, the condition lastGuess[msg.sender] == 0 checks if it's the player's first time playing indicating they haven't made any guess yet. The condition lastResetTime > lastGuess[msg.sender] checks if the secret number has been reset since the player's last guess. next line it resets the lastDifference for a player either when its their first guess or when the secret number has been reset since their last guess. This is needed for the games feedback mechanism ensuring accurate warmer or colder responses. I dont see it as a bug

Reply:
You're correct, I didn't read it properly. The `lastGuess[msg.sender] == 0` shouldn't be needed though. If everytime the games starts/resets the `lastResetTime` is updated then the players guess should be reset.

Reply:
Agreed; Removing the redundant condition check;